### PR TITLE
send project progress notifications #330

### PR DESCRIPTION
**Pull request by Hagellach37**
Hey @laurentS,
this one is for you. Will probably save you a lot of time in the future. :) You want to take a quick look?

### CODE REVIEW

File: mapswipe_workers/mapswipe_workers/utils/slack_helper.py [hunk 1]
```diff
@@ -27,14 +23,58 @@ def send_slack_message(message_type: str, project_name: str, project_id: str = N
             + "Make sure to activate the project using the manager dashboard.\n"
             + "Happy Swiping. :)"
         )
+        slack_client.chat_postMessage(channel=SLACK_CHANNEL, text=message)
     elif message_type == "fail":
```
Comment (laurentS) [inline mapswipe_workers/mapswipe_workers/utils/slack_helper.py:27]: You could use an [enum](https://docs.python.org/3/library/enum.html) for the `message_type`, it's a bit more structured than using `str`, and you get better type checking and syntax completion, which in turn prevents typos, etc...
```python
from enum import Enum
class MessageType(Enum):
    FAIL = 1
    NOTIFICATION_90 = 2
    ...

Then you can check with `if message_type == MessageType.FAIL`, etc...
You'll need to import the class everywhere you use it, but it's probably a good way to define these centrally.

**Comment by Hagellach37** [comment]
@laurentS thanks for the comments. I updated my code. :)

File: mapswipe_workers/mapswipe_workers/utils/slack_helper.py [hunk 2]
```diff
@@ -57,24 +57,26 @@ def send_progress_notification(project_id: int):
...
-    logger.info(
-        f"{project_id} - progress: {progress},"
-        f"notifications: {notification_90_sent} {notification_100_sent}"
-    )
 
-    if progress >= 90 and not notification_90_sent:
-        # send notification and set value in firebase
-        fb_db.reference(f"v2/projects/{project_id}/notification_90_sent").set(True)
-        send_slack_message("notification_90", project_name, project_id)
+    if progress >= 90:
```
Comment (laurentS) [inline mapswipe_workers/mapswipe_workers/utils/slack_helper.py:64]: I missed this the first time, but what data type is `fb_db.....get()` returning? I imagine a string? python might be unhappy about the comparison.

Comment (Matthias-Schaub) [reply mapswipe_workers/mapswipe_workers/utils/slack_helper.py:64]: .get() usually returns a list or ordered dict.

File: mapswipe_workers/mapswipe_workers/mapswipe_workers.py [hunk 3]
```diff
@@ -101,6 +104,7 @@ def run_firebase_to_postgres() -> list:
     update_data.update_user_data()
     update_data.update_project_data()
     project_ids = transfer_results.transfer_results()
+    [send_progress_notification(project_id) for project_id in project_ids]
```
Comment (Matthias-Schaub) [inline mapswipe_workers/mapswipe_workers/mapswipe_workers.py:107]: For me a simple for loop would be more readbale. Its unsual to do a list comprehension without saving the resulting list. At first it seems more like a potential bug.

File: mapswipe_workers/mapswipe_workers/utils/slack_helper.py (hunk 2 above)
Comment (Hagellach37) [reply mapswipe_workers/mapswipe_workers/utils/slack_helper.py:64]: in this case it returns an integer in this case, if if is a number in firebase.

**Pull request merged**